zng/builder.go

```diff
@@ -48,3 +54,168 @@ func (b *Builder) Build(zvs ...zcode.Bytes) *Record {
 	}
 	return &b.rec
 }
+
+// Parse creates a record from the zng string representation of each leaf field
+// in the DFS traversal of the record type.  If there aren't enough inputs values
+// to occupy every leaf value, then those values are left unset, in which case
+// a valid record is returned along with ErrIncomplete.
+// XXX This currently does not work for embedded sets, unions, and so forth.
+// We need to define a record value syntax and do proper recusive descent
+// parsing here.
+func (b *Builder) Parse(vals ...string) (*Record, error) {
```

Comment:
Nit: I think BuildFromStrings or something along that line would be a better name for this.